audformat/core/table.py

```diff
@@ -543,13 +543,117 @@ def _set_column(self, column_id: str, column: Column) -> Column:
         return column
 
 
+class MiscTable(Base):
+    r"""Miscellaneous table.
+
+    .. note:: Intended for use with tables
+        that have an index that is not conform to
+        :ref:`table specifications <data-tables:Tables>`.
+        Otherwise, use :class:`audformat.Table`.
+
+    To fill a table with labels,
+    add one or more :class:`audformat.Column`
+    and use :meth:`audformat.MiscTable.set` to set the values.
+
+    Args:
+        index: table index
+        split_id: split identifier (must exist)
+        media_id: media identifier (must exist)
+        description: database description
+        meta: additional meta fields
+
+    Example:
+        >>> index = pd.MultiIndex.from_tuples(
+        ...   [
+        ...     ('f1', 'f2'),
+        ...     ('f1', 'f3'),
+        ...     ('f2', 'f3'),
+        ...   ],
+        ...   names=['file', 'other'],
+        ... )
+        >>> table = MiscTable(
+        ...     index,
+        ...     split_id=define.SplitType.TEST,
+        ... )
+        >>> table['match'] = Column()
+        >>> table
+        levels: [file, other]
+        split_id: test
+        columns:
+          match: {}
+        >>> table.get()
+                   match
+        file other
+        f1   f2      NaN
+             f3      NaN
+        f2   f3      NaN
+        >>> table.set({'match': [True, False, True]})
+        >>> table.get()
+                   match
+        file other
+        f1   f2     True
+             f3    False
+        f2   f3     True
+        >>> table.get(index[:2])
+                   match
+        file other
+        f1   f2     True
+             f3    False
+
+    """
+    def __init__(
+            self,
+            index: pd.Index,
+            *,
+            split_id: str = None,
+            media_id: str = None,
+            description: str = None,
+            meta: dict = None,
+    ):
+        self.levels = None
+        r"""Index levels."""
+
+        super().__init__(
+            index,
+            split_id=split_id,
+            media_id=media_id,
+            description=description,
+            meta=meta,
+        )
+
+        if index is not None:
+            if isinstance(index, pd.MultiIndex):
+                levels = list(index.names)
+            else:
+                levels = [index.name]
+            self.levels = levels
+
+    def copy(self) -> 'MiscTable':
+        r"""Copy table.
+
+        Return:
+            new table object
+
+        """
+        return super().copy()
+
+    def _get_by_index(self, index: pd.Index) -> (pd.DataFrame, bool):
+        return self.df.loc[index], False
+
+    def _index_levels_and_converters(self) -> typing.Tuple[
+        typing.Sequence[str],
+        typing.Dict[str, typing.Callable],
+    ]:
+        return self.levels, {}
```

Comment:
We currently do not store type information for the levels. But since we cannot link schemes to index columns it might be useful to do so. @hagenw what's your opinion? In any case, I'd prefer to work on it in a separate PR.

Reply:
https://github.com/audeering/audformat/issues/190